sentry-android-replay/src/main/java/io/sentry/android/replay/capture/SessionCaptureStrategy.kt

```diff
@@ -78,6 +80,12 @@ internal class SessionCaptureStrategy(
             bitmap?.recycle()
             return
         }
+        val rateLimiter = hub?.rateLimiter
+        if (rateLimiter?.isActiveForCategory(All) == true || rateLimiter?.isActiveForCategory(Replay) == true) {
+            options.logger.log(DEBUG, "Skipping screenshot recording, rate limit is active")
+            bitmap?.recycle()
+            return
+        }
```

Comment:
@brustolin and @romtsn I think this is a different behaviour than on iOS https://github.com/getsentry/sentry-cocoa/pull/4496. On iOS, when a limit for SR gets active, we drop everything until the next session starts. Please align.

Reply:
Yes, in our last replay meeting, we agreed to start with the basics and stop session replay altogether for the remainder of the app session.

Reply:
>i'd prefer to do our best and start capturing from when the rate-limit is lifted?

That's also my opinion. This would also simplify the logic.

Reply:
But your current logic is not taking into account that the envelope with segment 0 could be the one that was lost. It does not matter anymore so send to following segments, they all will be ignored in the server.

Reply:
Btw, this approach is just throwing away the already-taken screenshot. Maybe Android is different; for iOS, making the screenshot is the most expensive part of the process.

Reply:
Another point to take into account, someone can correct me on this one, is that the most common form of rate limiting is quota exceeded. And this will only be lifted at the end of the month.

Reply:
> But your current logic is not taking into account that the envelope with segment 0 could be the one that was lost. It does not matter anymore to send to following segments, they all will be ignored in the server.

Why would it be lost? We're not capturing envelopes if rate limit is active right? Or do you mean if we get rate-limited upon sending the segment 0 event? In this case on Android it'd be stored in cache until rate-limit is lifted. The only case it gets lost if the cache gets rotated, but it's an edge-case enough to not handle it for now I guess.

> Btw, this approach is just throwing away the already-taken screenshot. Maybe Android is different; for iOS, making the screenshot is the most expensive part of the process.

That's a good point, I should probably move this logic to where we capture the screenshot 👍 

> Another point to take into account, someone can correct me on this one, is that the most common form of rate limiting is quota exceeded. And this will only be lifted at the end of the month.

Probably yes, but not sure about the end of the month (could be on-demand events etc etc).  I think this is also mostly relevant for big customers who have spike protection and whatnot in place, so we shouldn't think in "months" here probably :)

Reply:
> And this will only be lifted at the end of the month.

Nope, spike protection works on an hour basis: https://docs.sentry.io/pricing/quotas/spike-protection/#how-sentry-detects-spikes

Reply:
I don't have enough insight into SR. I let you two decide this.